.buildkite/scripts/snyk/report.sh

```diff
@@ -7,13 +7,6 @@ TARGET_BRANCHES=("main")
 # Resolves the branches we are going to track
 resolve_latest_branches() {
   source .buildkite/scripts/snyk/resolve_stack_version.sh
-  for SNAPSHOT_VERSION in "${SNAPSHOT_VERSIONS[@]}"
-  do
-    IFS='.'
-    read -a versions <<< "$SNAPSHOT_VERSION"
-    version=${versions[0]}.${versions[1]}
-    TARGET_BRANCHES+=("$version")
-  done
```

Comment:
These changes are good that simplify the logic. However, as my understanding, the CI agent has Java 17+ installed and out 7.17 branch ([gradle 7.2](https://github.com/elastic/logstash/blob/7.17/gradle/wrapper/gradle-wrapper.properties#L3)) is not compiling due to gradle vs Java versions mismatch: [compatibility matrix](https://docs.gradle.org/current/userguide/compatibility.html). 

It seems to me the resolution, we need (till 9.0 release) to put back install java and BK agent type changes from the [recent change](https://github.com/elastic/logstash/pull/16475/files).

Let me please know if I am missing anything.

Reply:
I'll followup with a fix for 7.17, in the meantime this PR just fixes the branch detection issue.